File: batch-service/domain/feed.go
package domain

import "time"

type Feed struct {
	ID                string
	Name              string
	RssURL            string
	TrendPlatformType TrendPlatformType
	Platform          Platform
	Category          Category
	CreatedAt         time.Time
	UpdatedAt         time.Time
	DeletedAt         *time.Time
}

type CreateFeedInputDTO struct {
	Name       string `json:"name"`
	RssURL     string `json:"rss_url"`
	PlatformID string `json:"platform_id"`
	CategoryID string `json:"category_id"`
}

type UpdateFeedInputDTO struct {
	ID         string `json:"id"`
	Name       string `json:"name"`
	RssURL     string `json:"rss_url"`
	PlatformID string `json:"platform_id"`
	CategoryID string `json:"category_id"`
}

File: web/admin/src/types/feed.ts
import { OriginArticleType } from "./article";
import { OriginCategoryType } from "./category";
import { OriginPlatformType } from "./platform";

export type OriginFeedType = {
  id: string;
  platformId: string;
  categoryId: string;
  name: string;
  description: string;
  rssUrl: string;
  siteUrl: string;
  thumbnailUrl: string;
  trendPlatformType: number;
  apiQueryParam?: string;
  createdAt: string;
  updatedAt: string;
  deletedAt?: string;
};

export type FeedType = OriginFeedType & {
  category: OriginCategoryType;
  platform: OriginPlatformType;
  articles: Array<OriginArticleType>;
};

export type FetchFeedsAPIResponse = {
  data: {
    feeds: Array<FeedType>;
    message: string;
  };
  status: number;
};
